### PR TITLE
Add always quote non empty

### PR DESCRIPTION
**Pull request by Richard-Lionheart**
Fixes #

## Proposed Changes

- Added new QuoteStrategy ALWAYS_NON_EMPTY, which quotes everything except empty or null fields 
- Added basic maven setup

## Performance

In order to prevent a performance degradation please run a benchmark
using `./gradlew jmh` before and after your change and show the results.

### CODE REVIEW

File: pom.xml
Comment (osiegmar) [inline pom.xml:1]: Please no changes to this file – see line no 1. I don't want to maintain it...

File: src/main/java/de/siegmar/fastcsv/writer/QuoteStrategy.java [hunk 1]
```diff
@@ -28,6 +28,10 @@ public enum QuoteStrategy {
     /**
      * Enclose any text field with quotes regardless of its content (even empty and {@code null} fields).
      */
-    ALWAYS
+    ALWAYS,
 
+    /**
+     * Enclose any text field with quotes if it has content, excluding empty and {@code null} fields.
+     */
+    ALWAYS_NON_EMPTY
```
Comment (osiegmar) [inline src/main/java/de/siegmar/fastcsv/writer/QuoteStrategy.java:36]: Please just name it `NON_EMPTY`.

File: pom.xml
Comment (Richard-Lionheart) [reply pom.xml:1]: done

File: src/main/java/de/siegmar/fastcsv/writer/QuoteStrategy.java (hunk 1 above)
Comment (Richard-Lionheart) [reply src/main/java/de/siegmar/fastcsv/writer/QuoteStrategy.java]: done

**Comment by osiegmar** [comment]
You need to rebase your changes on top of the develop brunch – currently 18 files are to be modified.

**Comment by Richard-Lionheart** [comment]
> You need to rebase your changes on top of the develop brunch – currently 18 files are to be modified.

I will create a new PR and close this one

**Pull request closed without merging**